client/components/Goals/DeleteGoalButton.tsx

```diff
@@ -0,0 +1,110 @@
+import { gql, useMutation } from '@apollo/client';
+import React, { FunctionComponent, useState } from 'react';
+import { FaTrashAlt } from 'react-icons/fa';
+import styled from 'styled-components';
+import { GET_GOALS_QUERY } from '../../pages/home';
+import Spinner from '../Spinner';
+import Modal from '../Utilities/Modal';
+
+const ButtonContainer = styled.div`
+  cursor: pointer;
+  margin-left: 10px;
+`;
+
+const Form = styled.form`
+  width: 100%;
+  max-width: 350px;
+  margin: 0 auto;
+`;
+
+const ConfirmMessage = styled.p`
+  font-size: 1.2rem;
+`;
+
+const GoalName = styled.p`
+  font-weight: 700;
+  margin-bottom: 50px;
+  text-transform: capitalize;
+`;
+
+const Note = styled.p`
+  font-size: 0.7rem;
+  text-transform: uppercase;
+`;
+
+const DELETE_GOAL_MUTATION = gql`
+  mutation($goalId: String!) {
+    deleteGoal(goalId: $goalId) {
+      deletedGoals
+      deletedTasks
+      deletedSubtasks
+    }
+  }
+`;
```

Comment:
This can be moved into `utils/graphql/mutation`. Also reference it in `mutation/index.js` to allow for named imports